lib/unexpected-immutable.js

```diff
@@ -24,6 +24,27 @@ module.exports = {
     expect.addAssertion('<Immutable> to [exhaustively] satisfy <any>', function (expect, subject, pattern) {
       return expect(subject.toJS(), 'to [exhaustively] satisfy', pattern)
     })
+    expect.addAssertion('<Immutable> extracting value at <array> <assertion>', function (expect, subject, path) {
+      expect(subject.hasIn(path), 'to be true')
+      return expect.shift(subject.getIn(path))
+    })
+    expect.addAssertion('<Immutable> extracting value at <string> <assertion>', function (expect, subject, pathStr) {
+      const path = pathStr.split(/[./]/)
+      return expect.shift(subject.getIn(path))
+    })
+    expect.addAssertion('<Immutable> to have deep value <array|string> <any?>', function (expect, subject, path, value) {
+      if (value !== undefined) {
+        expect.argsOutput = function (output) {
+          output.text('of ').appendInspected(value).text(' at ').appendInspected(path)
+        }
+        return expect(subject, 'extracting value at', path, 'to equal', value)
+      } else {
+        if (typeof path === 'string') {
+          path = path.split(/[./]/)
```

Comment:
Rather than reassigning path could you do sth like

const pathArray = typeof path === 'string' ? path.split(/[./]/) : path